plugin.py

```diff
@@ -531,6 +540,9 @@ def _update_stored_region_async(self) -> Tuple[bool, sublime.Region]:
         return False, sublime.Region(-1, -1)
 
     def on_modified_async(self) -> None:
+        if not self.is_valid_view():
+            return
+
         plugin = RustAnalyzer.plugin_from_view(self.view)
         if plugin is None:
             return
```

Comment:
Possible alternative:
```suggestion
        plugin = RustAnalyzer.plugin_from_view(self.view)
        if plugin is None:
            return

        session = plugin.weaksession()
        if not session or not session.session_view_for_view_async(self.view):
            return
```
(Ideally extracted into some common method)

Reply:
Or probably better to have:
```py
        session = plugin.weaksession()
        if not session or session.session_view_for_view_async(self.view):
            return
```
as a method of the plugin instance (`is_valid_view(view)`)

Reply:
Also, the `self.view.syntax().scope` will crash on Find Results view for example.

Reply:
I think this definition makes more sense. Thanks!